### PR TITLE
DRILL-5056: UserException does not write full message to log

### PR DESCRIPTION
**Pull request by paul-rogers**
A case occurred in which the External Sort failed during spilling. All
that was written to the log was:

2016-11-18 ... INFO  o.a.d.e.p.i.xsort.ExternalSortBatch - User Error
Occurred

Modified the logging code to provide more information to aid tracking
down problems that occur in the field.

Also cleaned up some warnings found while tracking down this issue.

The test case for this is {{TestSortSpillWithException}}, then look at the log output.

### CODE REVIEW

File: common/src/main/java/org/apache/drill/common/exceptions/UserException.java [hunk 1]
```diff
@@ -37,6 +37,7 @@
  *
  * @see org.apache.drill.exec.proto.UserBitShared.DrillPBError.ErrorType
  */
+@SuppressWarnings("serial")
```
Comment (Ben-Zvi) [inline common/src/main/java/org/apache/drill/common/exceptions/UserException.java]: Should there instead be in this class :
     private static final long serialVersionUID = -3796081521525479249L;
as does its super-class DrillRuntimeException, and as does its super-class RuntimeException, and as does its super-class Exception, and as does its super-class Throwable (which implements Serializable ........) ??
(Not sure how to pick the specific UID number ...)

File: common/src/main/java/org/apache/drill/common/exceptions/UserException.java [hunk 2]
```diff
@@ -549,7 +550,12 @@ public UserException build(final Logger logger) {
       if (isSystemError) {
         logger.error(newException.getMessage(), newException);
       } else {
-        logger.info("User Error Occurred", newException);
+        String msg = "User Error Occurred";
+        if (message != null) {
+          msg += ": " + message; }
+        if (cause != null) {
```
Comment (Ben-Zvi) [inline common/src/main/java/org/apache/drill/common/exceptions/UserException.java]: Wonder if the root cause (see line 540 above) may also be useful (in case it is below/different from 'cause') ?

File: common/src/main/java/org/apache/drill/common/exceptions/UserException.java (hunk 1 above)
Comment (paul-rogers) [inline common/src/main/java/org/apache/drill/common/exceptions/UserException.java]: There are three ways to handle serializable types. 1) assign a number (start with 1 and increase, say). 2. Let an IDE generate one, such as the one you mentioned. or 3) state that the class is never serialized, so don't worry about the ID, which is the approach I took. Drill never uses Java serialization so the serial ID is unused and unnecessary.

File: common/src/main/java/org/apache/drill/common/exceptions/UserException.java [hunk 3]
```diff
@@ -549,7 +550,12 @@ public UserException build(final Logger logger) {
       if (isSystemError) {
         logger.error(newException.getMessage(), newException);
       } else {
-        logger.info("User Error Occurred", newException);
+        String msg = "User Error Occurred";
```
Comment (sudheeshkatkam) [inline common/src/main/java/org/apache/drill/common/exceptions/UserException.java]: Use StringBuilder?

File: common/src/main/java/org/apache/drill/common/exceptions/UserException.java [hunk 4]
```diff
@@ -549,7 +550,12 @@ public UserException build(final Logger logger) {
...
+        if (message != null) {
+          msg += ": " + message; }
```
Comment (sudheeshkatkam) [inline common/src/main/java/org/apache/drill/common/exceptions/UserException.java]: `}` on new line, here and below.

File: exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentExecutor.java [hunk 5]
```diff
@@ -224,6 +224,7 @@ public void run() {
           ImpersonationUtil.getProcessUserUGI();
 
       queryUserUgi.doAs(new PrivilegedExceptionAction<Void>() {
+        @Override
```
Comment (sudheeshkatkam) [inline exec/java-exec/src/main/java/org/apache/drill/exec/work/fragment/FragmentExecutor.java]: Please undo these overrides. Although useful, these are unrelated to the issue. Maybe another ticket?

**Comment by paul-rogers** [comment]
Used a generated serial id for UserException (even though it is never serialized.) Remove the code cleanup. Changed code to use a string builder.

**Comment by sudheeshkatkam** [comment]
+1
